src/idl_parser.cpp

```diff
@@ -2596,6 +2600,20 @@ CheckedError Parser::ParseProtoFields(StructDef *struct_def, bool isextend,
         }
         EXPECT(']');
       }
+
+      if(type.enum_def && type.base_type != BASE_TYPE_UNION && field->value.constant == "0") {
+          if(!type.enum_def->FindByValue("0")) {
+              // Enum doesn't contain an implicit default, and a default was not
```

Comment:
Could you explain this code (conditions) and add a unit-tests for this code (min_ev==nullptr and min_e!=nullptr)?
Is this case possible with a correctly defined .proto?

Reply:
* `if(type.enum_def && type.base_type != BASE_TYPE_UNION && field->value.constant == "0")` - a field of enum type without default value.
* `type.enum_def->FindByValue("0")` - the enum type doesn't have 0-value enumerator.

```proto
message Proto {
  enum Label { X = 1; } 
  optional Label label = 1;
}
```
Generated fbs-schema will be correct. Why this case is exceptional?